.cursor/rules/11-facets.mdc

```diff
@@ -0,0 +1,16 @@
+---
+name: Diamond facets
+description: Facet-only requirements and validations
+globs:
+  - 'src/Facets/**/*.sol'
+---
+
+@15-architecture.mdc
+@16-security.mdc
+@17-gas.mdc
+
+- Required functions: `_startBridge` (internal), `swapAndStartBridgeTokensVia{FacetName}`, `startBridgeTokensVia{FacetName}`; attach `nonReentrant`, `refundExcessNative`, `validateBridgeData`, and appropriate does/don'tContain swap/call modifiers.
```

Comment:
I would also add a rule abou constructor arguments that zero values should not be allowed. Code should validate addresses against the zero address and numeric parameters against zero

Reply:
yes, I agree. The GenericSwapFacetV3 for example has a nativeAddress parameter that in fact is address(0) on most networks